sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringTypeUtils.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+import org.apache.spark.{SparkContext, SparkEnv}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.internal.SqlApiConf
+
+object StringTypeUtils extends Logging {
```

Comment:
Do we still need this? My preference is to avoid premature perf optimization until we have numbers to prove we need to improve the perf.

Reply:
When we use collation internally we need collationId, but the config is a collationName. I am just not sure where to include this conversion then. We also want to make sure the conversion is not called on execution side.

Reply:
We can make `SqlApiConf.get.defaultCollation` to return the collation id, and move the implementation there.

Reply:
or even further, add `SqlApiConf.get.defaultStringType` which returns `StringType`